components/eventing-controller/testing/test_helpers.go

```diff
@@ -0,0 +1,269 @@
+			GrantType:    "client_credentials",
+			ClientId:     "xxx",
+			ClientSecret: "xxx",
+			TokenUrl:     "https://oauth2.xxx.com/oauth2/token",
+			Scope:        []string{"guid-identifier"},
+		},
+	}
+}
+
+func WithFilter(s *eventingv1alpha1.Subscription) {
+	s.Spec.Filter = &eventingv1alpha1.BebFilters{
+		Dialect: "beb",
+		Filters: []*eventingv1alpha1.BebFilter{
+			{
+				EventSource: &eventingv1alpha1.Filter{
+					Type:     "exact",
+					Property: "source",
+					Value:    "/default/kyma/myinstance",
+				},
+				EventType: &eventingv1alpha1.Filter{
+					Type:     "exact",
+					Property: "type",
+					Value:    "kyma.ev2.poc.event1.v1",
+				},
+			},
+		},
+	}
+}
+
+func WithValidSink(svcNs, svcName string, s *eventingv1alpha1.Subscription) {
+	s.Spec.Sink = fmt.Sprintf("https://%s.%s.svc.cluster.local", svcName, svcNs)
+}
+
+// fixtureValidSubscription returns a valid subscription
+func FixtureValidSubscription(name, namespace, id string) *eventingv1alpha1.Subscription {
+	return &eventingv1alpha1.Subscription{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Subscription",
+			APIVersion: "eventing.kyma-project.io/v1alpha1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: eventingv1alpha1.SubscriptionSpec{
+			ID:       id,
+			Protocol: "BEB",
+			ProtocolSettings: &eventingv1alpha1.ProtocolSettings{
+				ContentMode:     eventingv1alpha1.ProtocolSettingsContentModeBinary,
+				ExemptHandshake: true,
+				Qos:             "AT-LEAST_ONCE",
+				WebhookAuth: &eventingv1alpha1.WebhookAuth{
+					Type:         "oauth2",
+					GrantType:    "client_credentials",
+					ClientId:     "xxx",
+					ClientSecret: "xxx",
+					TokenUrl:     "https://oauth2.xxx.com/oauth2/token",
+					Scope:        []string{"guid-identifier"},
+				},
+			},
+			Sink: fmt.Sprintf("https://webhook.%s.svc.cluster.local", namespace),
+			Filter: &eventingv1alpha1.BebFilters{
+				Dialect: "beb",
+				Filters: []*eventingv1alpha1.BebFilter{
+					{
+						EventSource: &eventingv1alpha1.Filter{
+							Type:     "exact",
+							Property: "source",
+							Value:    "/default/kyma/myinstance",
+						},
+						EventType: &eventingv1alpha1.Filter{
+							Type:     "exact",
+							Property: "type",
+							Value:    "kyma.ev2.poc.event1.v1",
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func NewSubscriberSvc(name, ns string) *corev1.Service {
+	return &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: ns,
+		},
+		Spec: corev1.ServiceSpec{
+			Ports: []corev1.ServicePort{
+				{
+					Protocol: "TCP",
+					Port:     443,
+					TargetPort: intstr.IntOrString{
+						IntVal: 8080,
+					},
+				},
+			},
+			Selector: map[string]string{
+				"test": "test",
+			},
+		},
+	}
+}
+
+func SetSinkSvcPortInAPIRule(apiRule *apigatewayv1alpha1.APIRule, sink string) {
+	sinkURL, err := url.ParseRequestURI(sink)
+	if err != nil {
+		log.Fatalf("failed to parse sink URI: %v", err)
```

Comment:
Since we use this code in ginkgo tests, I think we should use an assertion here instead or even better return an error and do the assertion outside of the function.

```suggestion
		Expect(err).ShouldNot(HaveOccurred())
```

Reply:
Done.